subsys/pm/power.c

```diff
@@ -232,6 +232,8 @@ enum pm_state pm_system_suspend(int32_t ticks)
 	case PM_STATE_SUSPEND_TO_RAM:
 		__fallthrough;
 	case PM_STATE_SUSPEND_TO_DISK:
+		__fallthrough;
+	case PM_STATE_SOFT_OFF:
```

Comment:
I guess it's ok for now, but since entering `PM_STATE_SOFT_OFF` implies a later system reboot (full context loss), peripherals may be put first in `SUSPEND` state and then in the `OFF` state since in theory they'll consume even less power. This is another area that needs some more clarification, specially because it relies on `pm_low_power_devices` and `pm_suspend_devices` which, at least in-tree, do not have a consistent behavior.

Reply:
yes .... Can you approve it ?

Reply:
I won't block, but I'm not sure if this is really solving the problem, so I prefer other reviewers to vote on this one. The mapping between device PM states and system PM states is not clear to me, as noted in my previous comment.